### ISSUE TITLE
[R-package] early_stopping_round can be passed to lightgbm if verbose is different than 0

### ISSUE THREAD

**Issue by DoruSonic** [microsoft/LightGBM#5374]
If the parameter early_stopping_round is set and verbose is 1 or 2, even without validation data, it is passed without any error. As soon as verbose is set to 0, it outputs the error below and the model is not trained.

> Error in init(env = env) : 
>  For early stopping, valids must have at least one element

## Reproducible example taken from the tutorial

```r
library(lightgbm)
library(methods)

data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")
train <- agaricus.train
test <- agaricus.test

train_params <- list(
  num_leaves = 4L,
  learning_rate = 1.0,
  objective = "binary",
  nthread = 2L,
  num_iterations = 100,
  early_stopping_round = 10
)

bst <- lightgbm(
  data = as.matrix(train$data)
  , params = train_params
  , label = train$label
  , nrounds = 2L
  , verbose = 0L
)
```

## Environment info

LightGBM version or commit hash: 3.3.2

Command(s) you used to install LightGBM:

```r
install.packages("lightgbm", repos = "https://cran.r-project.org")
```

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

This is a known issue in v3.3.2. If you provided `verbose > 0` to `lightgbm::lightgbm()`, the training data will automatically be added a validation set. If you pass `verbose <= 0`, that doesn't happen, which leads to this expected error.

See https://github.com/microsoft/LightGBM/issues/5347#issuecomment-1175454082 for more details and for an approach that works as expected on both v3.3.2 and the latest development version in the repo.

-----

I suspect you may be new to GitHub. I've edited your description a bit so that code is formatted as code. You can learn how to do this (and a lot more!) at https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code.
